hr_payroll_commission/models/hr_payslip.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# © 2016 Coninckx David (Open Net Sarl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from openerp import models, fields, api
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    # ---------- Fields management
+
+    invoices = fields.One2many('account.invoice', 'slip_id',
+                               string='Invoices')
+    move_lines = fields.One2many('account.move.line', 'slip_id',
+                                 string='Journal Items')
+
+    # ---------- Utilities
+
+    @api.multi
+    def compute_sheet(self):
+
+        # First, detach invoices from the pay slips
+        InvoiceObj = self.env['account.invoice']
+        invoices = InvoiceObj.search([('slip_id', 'in', self.ids)])
+        if invoices:
+            invoices.write({'slip_id': False})
+
+        # Second, detach account move lines from the pay slips
+        AccountMoveLineObj = self.env['account.move.line']
+        aml = AccountMoveLineObj.search([('slip_id', 'in', self.ids)])
+        if aml:
+            aml.write({'slip_id': False})
```

Comment:
First, your method is too long. Seperate it in pieces. Second, what you are doing is very confusing. It seems a very bizare workflow.